OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnPrerequisite.cs

```diff
@@ -62,6 +61,11 @@ void INotifyRemovedFromWorld.RemovedFromWorld(Actor self)
 				globalManager.Unregister(self, this, info.Prerequisites);
 		}
 
+		void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newOwner)
+		{
+			globalManager = newOwner.PlayerActor.Trait<GrantConditionOnPrerequisiteManager>();
```

Comment:
This is done by `INotifyRemovedFromWorld.RemovedFromWorld` (capturing an actor removes it from the world, then changes the owner, and then adds it back to the world).